packages/cache-inmemory/src/services/inmemory-cache.ts

```diff
@@ -0,0 +1,107 @@
+import {
+  ConfigurableModuleDeclaration,
+  MODULE_RESOURCE_TYPE,
+  TransactionBaseService,
+  ICacheService,
+} from "@medusajs/medusa"
+import { MedusaError } from "medusa-core-utils"
+import { EntityManager } from "typeorm"
+
+import { CacheRecord, InMemoryCacheModuleOptions } from "../types"
+
+const DEFAULT_TTL = 30 // seconds
+
+type InjectedDependencies = {}
+
+/**
+ * Class represents basic, in-memory, cache store.
+ */
+class InMemoryCacheService
+  extends TransactionBaseService
+  implements ICacheService
+{
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+
+  protected TTL: number
+
+  protected readonly timoutRefs = {}
```

Comment:
todo: use Map instead for dynamic keys it is better and more performant, specially for a lot of values